ts/components/wallet/layout/WalletLayout.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * Wallet layout, split into two parts:
+ * - the top-most one shows a header w/ a title, subtitle,
+ *   image and touchable content (or a subset of these)
+ * - the bottom-most part shows the current screen's contents
+ *   (e.g. list of payment methods)
+ */
+
+import { Body, Button, Container, Icon, Text, View } from "native-base";
+import * as React from "react";
+import { Grid, Row } from "react-native-easy-grid";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import I18n from "../../../i18n";
+
+import { Left } from "native-base";
+import { WalletStyles } from "../../styles/wallet";
+import AppHeader from "../../ui/AppHeader";
+import { TOP_CONTENTS_MAX_SIZE, TopContents } from "./TopContents";
+import { TopContent } from "./types";
+
+export enum ImageType {
+  BANK_IMAGE
+}
+
+type Props = Readonly<{
+  headerTitle: string; // header to be shown in the AppHeader (may differ from "title")
+  allowGoBack: boolean; // whether the "back" button should be displayed
+  title: string;
+  topContent: TopContent;
+  children?: React.ReactElement<any>;
+  rightImage?: ImageType;
+  navigation: NavigationScreenProp<NavigationState>;
+}>;
+
+// size of the bottom part of the screen
+// (i.e. the actual contents passed in this.props.children)
+const WALLET_LAYOUT_BOTTOM_SIZE = 4;
+export class WalletLayout extends React.Component<Props, never> {
+  private payNoticeButton(): React.ReactNode {
+    return (
+      <View footer={true}>
+        <Button block={true}>
+          <Icon type="FontAwesome" name="qrcode" />
+          <Text>{I18n.t("wallet.payNotice")}</Text>
+        </Button>
+      </View>
+    );
+  }
+
+  /**
+   * The top part is defined by the TopContent component,
+   * the bottom part is defined by this.props.children
+   */
+  private twoPartsLayout(): React.ReactNode {
+    return (
+      <Grid>
+        <Row size={TopContents.getSize(this.props.topContent)}>
+          <TopContents {...this.props} />
+        </Row>
+        <Row size={this.getBottomSize()}>{this.props.children}</Row>
+      </Grid>
+    );
+  }
+
+  private getBottomSize() {
+    /**
+     * from the default bottom size, add the
+     * size that is not being used by the top
+     * part (either because subtitles or touchable
+     * parts are not used). The "unused" part
+     * is given by MAX_TOP_PART - ACTUAL_TOP_PART
+     */
+    return (
+      WALLET_LAYOUT_BOTTOM_SIZE +
+      (TOP_CONTENTS_MAX_SIZE - TopContents.getSize(this.props.topContent))
+    );
+  }
+
+  private goBackButton(): React.ReactNode {
+    if (this.props.allowGoBack === true) {
+      return (
+        <Left>
+          <Button
+            transparent={true}
+            onPress={_ => this.props.navigation.goBack()}
+          >
+            <Icon style={WalletStyles.white} name="chevron-left" />
+          </Button>
+        </Left>
+      );
+    }
+    return (
+      <Left>
```

Comment:
Just return `<Left/>`

Reply:
The problem with just returning `<Left/>` is that the height of the header adapts to that of the Text (and not the `Button`), hence the `Button`
<img src="https://user-images.githubusercontent.com/107715/40373165-9b562eb6-5de6-11e8-9ce6-bda8b7c95cb5.png" width="256"/>
<img src="https://user-images.githubusercontent.com/107715/40373167-9bc8ea5a-5de6-11e8-8c29-01dbbf4f9633.png" width="256"/>

Reply:
I'll investigate about this but can you try using Title instead of Text inside the Header Body?

Reply:
Can do, but that changes the font size. Also, other screens are using `Text` (e.g. [IdpSelectionScreen](https://github.com/teamdigitale/italia-app/blob/a4f0fc27f36369c183b3e1eb5189dca3aecb9754/ts/screens/authentication/IdpSelectionScreen.tsx#L105). How should I proceed?

Reply:
@fgiobergia could you please open a story to fix this? I would avoid further blocking this PR for now